src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
+
+impl From<NulError> for NameToIndexError {
+    fn from(_: NulError) -> NameToIndexError {
+        NameToIndexError::NulError
+    }
+}
+
+impl From<io::Error> for NameToIndexError {
+    fn from(e: io::Error) -> NameToIndexError {
+        NameToIndexError::IOError(e)
+    }
+}
+
+extern {
+    fn if_nametoindex(ifname: *const c_char) -> c_uint;
+}
+
+/// Resolve an interface into a interface number.
+pub fn name_to_index(name: &str) -> Result<c_uint, NameToIndexError> {
```

Comment:
> A few things:
> 
> use the function's name as in C

Fixed in 2cb7878

> return a nix::Result<c_uint>, which is equivalent to swapping in nix::Error for the error type

Fixed in 5ef9689

> use a NixPath bound for the string argument (the name is bad, and it'll likely change soon, see #221 if you want more info!)

That currently seems a little messy and missing an implementation for ``&str``? Interface names are not Paths and I don't want to require passing in CStrings, because at that point it stops being a Rust-Interface, I think.

> The nix::from_ffi() function handles errno translation for you. :-)

If i am reading this right, won't ``from_ffi()`` just return ``()`` ? At least the compiler seems to agree with me. Additionally, it only works for ``c_int``, but ``if_nametoindex`` returns unsigned values.